packages/template-retail-react-app/app/commerce-api/einstein.js

```diff
@@ -53,6 +53,52 @@ class EinsteinAPI {
         return body
     }
 
+    /**
+     * Given a product source, returns the data that Einstein requires
+     */
+    _destructureProduct(product) {
+        if (product.type && (product.type.master || product.type.variant)) {
+            // handle variants for PDP / viewProduct
+            return {
+                id: product.master.masterId,
+                sku: product.id,
+                altId: '',
+                altIdType: ''
+            }
+        } else if (
+            // handle variants for PLP / viewCategory & viewSearch
+            product.productType &&
+            (product.productType.master || product.productType.variant)
+        ) {
+            return {
+                id: product.productId,
+                sku: product.productId, //TODO: Should this be product.representedProduct.id instead?
+                altId: '',
+                altIdType: ''
+            }
+        } else {
+            // handles non-variants
+            return {
+                id: product.id,
+                sku: '',
+                altId: '',
+                altIdType: ''
+            }
+        }
+    }
```

Comment:
Whie the PWA Kit only supports variant products at the moment, I tried to do a bit of future-proofing here to handle other product types.